ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -527,12 +527,66 @@ public func concat<T, E>(next: SignalProducer<T, E>)(producer: SignalProducer<T,
 	return SignalProducer(values: [producer, next]) |> concat
 }
 
+/// Merges a `producer` of SignalProducers down into a single producer, biased toward the producers
+/// added earlier. Returns a SignalProducer that will forward signals from the original producers
+/// as they arrive.
+public func merge<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+
+	let relay = SignalProducer<T, E> { relayObserver, relayDisposable in
+		
+		let inFlight = Atomic(1)
```

Comment:
I followed https://github.com/ReactiveCocoa/ReactiveCocoa/releases/tag/swift-before-signal-producers here... I don’t really have an opinion on this, but looks like other .swift implementations in the project use `Atomic` instead of `OSAtomicXXX`.